src/server/mod.rs

```diff
@@ -70,31 +85,40 @@ impl SpaceapiServerBuilder {
     }
 
     pub fn build(self) -> Result<SpaceapiServer, SpaceapiServerError> {
-        // Log some useful debug information
-        debug!("Redis connection info: {:?}", &self.redis_connection_info);
-
-        // Create redis pool
-        let redis_config = r2d2::Config::builder()
-            // Provide up to 6 connections in connection pool
-            .pool_size(6)
-            // At least 1 connection must be active
-            .min_idle(Some(2))
-            // Initialize connection pool lazily. This allows the SpaceAPI
-            // server to work even without a database connection.
-            .initialization_fail_fast(false)
-            // Try to get a connection for max 1 second
-            .connection_timeout(Duration::from_secs(1))
-            // Don't log errors directly.
-            // They can get quite verbose, and we're already catching and
-            // logging the corresponding results anyways.
-            .error_handler(Box::new(r2d2::NopErrorHandler))
-            .build();
-        let redis_manager = RedisConnectionManager::new(self.redis_connection_info?)?;
-        let pool = r2d2::Pool::new(redis_config, redis_manager)?;
+
+        let pool = match self.redis_info {
+            RedisInfo::None => Err("No redis connection defined".into()),
+            RedisInfo::Err(e) => Err(e),
+            RedisInfo::Pool(p) => Ok(p),
+            RedisInfo::ConnectionInfo(ci) => {
+                // Log some useful debug information
+                debug!("Connecting to redis database {} at {:?}",
+                       ci.db, ci.addr);
+
+                // Create redis pool
+                let redis_config = r2d2::Config::builder()
+                    // Provide up to 6 connections in connection pool
+                    .pool_size(6)
+                    // At least 1 connection must be active
+                    .min_idle(Some(2))
+                    // Initialize connection pool lazily. This allows the SpaceAPI
+                    // server to work even without a database connection.
+                    .initialization_fail_fast(false)
+                    // Try to get a connection for max 1 second
+                    .connection_timeout(Duration::from_secs(1))
+                    // Don't log errors directly.
+                    // They can get quite verbose, and we're already catching and
+                    // logging the corresponding results anyways.
+                    .error_handler(Box::new(r2d2::NopErrorHandler))
+                    .build();
+                let redis_manager = RedisConnectionManager::new(ci)?;
+                Ok(r2d2::Pool::new(redis_config, redis_manager)?)
+            }
+        };
 
         Ok(SpaceapiServer {
             status: self.status,
-            redis_pool: pool,
+            redis_pool: pool.expect("Error initializing redis pool"),
```

Comment:
Could you return the error instead by just using `pool?` instead of panicking with `expect`? Alternatively you could already return appropriately in the match statement above.

Reply:
oh, right, `redis_pool` is a result. Might as well use it :)

Reply:
Wait, disregard that :D